src/System.Memory/tests/ReadOnlySpan/IndexOfAny.char.cs

```diff
@@ -0,0 +1,519 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class ReadOnlySpanTests
+    {
+        [Theory]
+        [InlineData("a", "a", 'a', 0)]
+        [InlineData("ab", "a", 'a', 0)]
+        [InlineData("aab", "a", 'a', 0)]
+        [InlineData("acab", "a", 'a', 0)]
+        [InlineData("acab", "c", 'c', 1)]
+        [InlineData("abcdefghijklmnopqrstuvwxyz", "lo", 'l', 11)]
+        [InlineData("abcdefghijklmnopqrstuvwxyz", "ol", 'l', 11)]
+        [InlineData("abcdefghijklmnopqrstuvwxyz", "ll", 'l', 11)]
+        [InlineData("abcdefghijklmnopqrstuvwxyz", "lmr", 'l', 11)]
+        [InlineData("abcdefghijklmnopqrstuvwxyz", "rml", 'l', 11)]
+        [InlineData("abcdefghijklmnopqrstuvwxyz", "mlr", 'l', 11)]
+        [InlineData("abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz", "lmr", 'l', 11)]
+        [InlineData("aaaaaaaaaaalmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz", "lmr", 'l', 11)]
+        [InlineData("aaaaaaaaaaacmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz", "lmr", 'm', 12)]
+        [InlineData("aaaaaaaaaaarmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz", "lmr", 'r', 11)]
+        [InlineData("/localhost:5000/PATH/%2FPATH2/ HTTP/1.1", " %?", '%', 21)]
+        [InlineData("/localhost:5000/PATH/%2FPATH2/?key=value HTTP/1.1", " %?", '%', 21)]
+        [InlineData("/localhost:5000/PATH/PATH2/?key=value HTTP/1.1", " %?", '?', 27)]
+        [InlineData("/localhost:5000/PATH/PATH2/ HTTP/1.1", " %?", ' ', 27)]
+        public static void IndexOfAnyStrings_Char(string raw, string search, char expectResult, int expectIndex)
+        {
+            var span = raw.AsSpan();
+            char[] searchFor = search.ToCharArray();
+
+            var index = -1;
+            if (searchFor.Length == 1)
+            {
+                index = span.IndexOf(searchFor[0]);
+            }
+            else if (searchFor.Length == 2)
+            {
+                index = span.IndexOfAny(searchFor[0], searchFor[1]);
+            }
+            else if (searchFor.Length == 3)
+            {
+                index = span.IndexOfAny(searchFor[0], searchFor[1], searchFor[2]);
+            }
+            else
+            {
+                index = span.IndexOfAny(searchFor);
+            }
+
+            var found = span[index];
+            Assert.Equal(expectResult, found);
+            Assert.Equal(expectIndex, index);
+        }
+
+        [Fact]
+        public static void ZeroLengthIndexOfTwo_Char()
+        {
+            ReadOnlySpan<char> sp = new ReadOnlySpan<char>(Array.Empty<char>());
+            int idx = sp.IndexOfAny<char>((char)0, (char)0);
+            Assert.Equal(-1, idx);
+        }
+
+        [Fact]
+        public static void DefaultFilledIndexOfTwo_Char()
```

Comment:
I'm not sure what is this test testing.

Reply:
Yep, can't make sense of it anymore. I will have to dig up what the test used to be.

I think the intention was to look for (default, x) and (x, default) in a span containing only default, and assert we always find it at the first index.

Therefore, the inner loop is unnecessary, at least the way the test is written atm:
`for (int i = 0; i < length; i++)`

It should just be single check. The test should look like this:
```C#
            for (int length = 0; length < byte.MaxValue; length++)
            {
                byte[] a = new byte[length];
                Span<byte> span = new Span<byte>(a);

                 int idx = span.IndexOfAny(default, 99);
                 Assert.Equal(0, idx);

                 idx = span.IndexOfAny(99, default);
                 Assert.Equal(0, idx);
            }
```

We can leave it as is for now and fix all the tests (byte/char/etc.) in one go outside of this PR.